engine/promise_test.go

```diff
@@ -73,3 +73,42 @@ func TestPromise_Force(t *testing.T) {
 		assert.Equal(t, 10, count)
 	})
 }
+
+func TestPromise_ForceWithDelayedSequenceExecutions(t *testing.T) {
+	var res []int
+	k := DelaySeq(
+		func() NextFunc {
+			i := 0
+			return func() (PromiseFunc, bool) {
+				defer func() { i++ }()
+
+				v := i
+				res = append(res, v)
+				return func(ctx context.Context) *Promise {
+					return Bool(v%((v%4)+1) == 0)
+				}, i < 11
+			}
+		}())
+
+	t.Run("ok", func(t *testing.T) {
+		cases := []struct {
+			wantOk  bool
+			wantRes []int
+		}{
+			{wantOk: true, wantRes: []int{0}},
+			{wantOk: true, wantRes: []int{1, 2, 3, 4}},
+			{wantOk: true, wantRes: []int{5, 6}},
+			{wantOk: true, wantRes: []int{7, 8}},
+			{wantOk: false, wantRes: []int{9, 10, 11}},
+			{wantOk: false, wantRes: nil},
+		}
+
+		for _, tc := range cases {
+			res = nil
+			ok, err := k.Force(context.Background())
+			assert.NoError(t, err)
+			assert.Equal(t, tc.wantOk, ok)
+			assert.Equal(t, tc.wantRes, res)
+		}
+	})
+}
```

Comment:
I don't see the point. `k` is initialized once because we are testing the evolution of its state across multiple calls (`.Force`). And it's the same for `var res []int`.